### PR TITLE
Added IsStatusWrongInputWithReason

### PR DESCRIPTION
**Pull request by ewoutp**
Added functions to query for WrongInput errors with a specific reason.

### CODE REVIEW

**Comment by ewoutp** [comment]
@zyndiecate RFR

File: util.go [hunk 1]
```diff
@@ -79,7 +88,14 @@ func isStatus(statusCode int, responseBody string) (bool, error) {
 	}
 
 	if responsePayload.StatusCode == statusCode {
-		return true, nil
+		if reason == "" {
+			// We're not looking for a specific reason, so we're done
+			return true, nil
+		}
+		// Match end of status text to ": " + <reason>
+		if strings.HasSuffix(responsePayload.StatusText, ": "+reason) {
```
Comment (zyndiecate) [inline util.go]: I would like to split `responsePayload.StatusText` by `:` and see if everything after the colon is exactly the same as `reason`.

**Comment by zyndiecate** [comment]
When seeing this I see we miss tests here completely, but this is another issue.

LGTM

File: util.go (hunk 1 above)
Comment (ewoutp) [inline util.go]: I've wanted to do that, but there is no requirement for either text or reason NOT to contain a colon.

That means that there can be many parts and you do not know which to compare.

**Pull request merged**